### PR TITLE
[System] Fixes HttpListenerRequest.IsLocal.

### PR DESCRIPTION
**Pull request by esdrubal**
IsLocal was only returning true when the loopback address was used.
Fixes [#38322](https://bugzilla.xamarin.com/show_bug.cgi?id=38322).

### CODE REVIEW

File: mcs/class/System/System.Net/HttpListenerRequest.cs [hunk 1]
```diff
@@ -403,7 +403,7 @@ internal bool FlushInput ()
 		}
 
 		public bool IsLocal {
-			get { return IPAddress.IsLoopback (RemoteEndPoint.Address); }
+			get { return LocalEndPoint.Address.Equals (RemoteEndPoint.Address); }
```
Comment (grendello) [inline mcs/class/System/System.Net/HttpListenerRequest.cs]: I'm not sure this is a complete solution. If the device has more than one interface it is possible that the connection will be between two different addresses which are still local to the machine, however. I think that RemoteEndPoint.Address has to be checked against the list of interfaces on the local machine.

Comment (esdrubal) [inline mcs/class/System/System.Net/HttpListenerRequest.cs]: Reference source is doing it this way.
http://referencesource.microsoft.com/#System/net/System/Net/HttpListenerRequest.cs,458

When connecting to an endpoint the remote endpoint is specified but the local endpoint is not. I don't know if there is a way of connecting to a local endpoint through a different local endpoint.

Comment (esdrubal) [inline mcs/class/System/System.Net/HttpListenerRequest.cs]: So in a windows virtual machine with 2 bridge adapters to the same network. When we have a HttpListenerRequest with LocalEndPoint equal to 192.168.1.103:9000 and RemoteEndPoint equal to 192.168.1.104:9000 HttpListenerRequest.IsLocal returns false.

Comment (grendello) [inline mcs/class/System/System.Net/HttpListenerRequest.cs]: OK, if we're still in for bug-to-bug compatibility, then I guess it's fine. I still don't think this behavior is correct, though :)

**Comment by alexrp** [comment]
LGTM

**Pull request merged**